### PR TITLE
Role Feature Flags

### PR DESCRIPTION
**Pull request by tbezman**
Pretty simple, made a new map of `Role => Flag => Boolean`. 

I also updated the `isFeatureEnabled` function to return `true` if any of the checks are `true`. This fixes the annoying situation where I update `PROD_FLAGS` but don't update `EMPLOYEE_FLAGS` and then none of the employees at Gallery can access it

Re: the schema changes
I added a script to fetch the latest schema from `main`. this should stop people's editors from formatting the file. 

In this PR, I just copied the file directly and I didn't see any formatting happening. So I'm not sure who the culprit is 🕵️

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "lint": "eslint . --ext .js,.ts,.tsx,.cjs",
     "typecheck": "nextjs-routes; graphql-codegen; tsc --noEmit",
     "relay": "mkdir __generated__; relay-compiler",
+    "fetch-schema": "curl https://raw.githubusercontent.com/gallery-so/go-gallery/main/graphql/schema/schema.graphql > schema.graphql",
```
Comment (Robinnnnn) [inline package.json:10]: 😍

File: src/utils/graphql/isFeatureEnabled.tsx [hunk 2]
```diff
@@ -1,36 +1,45 @@
...
+  ADMIN: {
+    WHITE_RHINO: true,
+    EMAIL: true,
+  },
+  BETA_TESTER: {
+    WHITE_RHINO: true,
+    EMAIL: true,
+  },
+  // In case a new role gets added, how do we want to handle that new role
+  // if the client hasn't been updated to handle the new role.
```
Comment (Robinnnnn) [inline src/utils/graphql/isFeatureEnabled.tsx:36]: lazy solution: deploy backend first then update frontend

longer term: have backend return specific feature flags

File: src/utils/graphql/isFeatureEnabled.tsx [hunk 3]
```diff
@@ -1,36 +1,45 @@
...
 };
 
+const ROLE_FLAGS: Record<Role, Record<FeatureFlag, boolean>> = {
+  ADMIN: {
+    WHITE_RHINO: true,
+    EMAIL: true,
+  },
+  BETA_TESTER: {
+    WHITE_RHINO: true,
+    EMAIL: true,
```
Comment (Robinnnnn) [inline src/utils/graphql/isFeatureEnabled.tsx:33]: let's keep `EMAIL: false` for beta testers

**Review by Robinnnnn — APPROVED** [review]

File: src/utils/graphql/isFeatureEnabled.tsx (hunk 2 above)
Comment (tbezman) [reply src/utils/graphql/isFeatureEnabled.tsx:31]: @Robinnnnn This comment is trying to explain that "the below object is what will be used in the case that the backend deployed something that the frontend doesn't know about"

I'm not literally asking a question, maybe I should change my language here?

**Pull request merged**